### ISSUE TITLE
QR Code Scanner

### ISSUE THREAD

**Issue by hitchcott** [getumbrel/umbrel#355]
It would be convenient to be able to scan QR codes using a JS code scanner via webcam.

A modern and lightweight implementation with no deps such as https://github.com/nimiq/qr-scanner could be used.

This could be used for bitcoin withdrawals and lightning invoice payments, and perhaps other data entry.

I would be happy to submit a PR if this is desirable.

**Comment by jackgopack4** [comment]
Seconding this request!

**Comment by mayankchhabra** [comment]
Yeah that would have been really cool, but can't really support it unfortunately. Umbrel's dashboard is served over HTTP on the local network, along with remote Tor access — none of which support reading from the camera stream.

**Comment by hitchcott** [comment]
I see, yeah this wouldn't be possible with the default setup.

But it did get me thinking:

- Could there be contexts where the user is hosting this behind a proxy that supports HTTPS?
- Also, would this be possible if a self signed certificate was used?
- This limitation might have knock on effects on the functionality of future umbrel apps. Is there a potential workaround for power users?

**Comment by hitchcott** [comment]
Related message in telegram:


> Any plans to support SSL with specter?  Right now no camera access (QR) due to no HTTPS correct?

**Comment by maltokyo** [comment]
Run umbrel behind reverse proxy fixes this, I have it all (including apps) running with https.

See here: https://github.com/getumbrel/umbrel/issues/406

**Comment by hitchcott** [comment]
To add to the previous list, another scenario might be where an internal network has its own CA.

In any case, a self-signed cert should probably be enabled for security anyway; theres probably malicious IOT devices on people's network sniffing traffic and harvesting passwords, which a self-signed cert would defend against; not completely, but a more targeted MITM attack would be required.

Also, a relevant thread about HTTPS for local networks: https://news.ycombinator.com/item?id=23669051

**Comment by hitchcott** [comment]
Seems like some TOR browsers will be allowing secure context apis:

https://github.com/brave/brave-browser/issues/1135
